src/main/java/net/minecraftforge/common/brewing/IBrewingRecipe.java

```diff
@@ -19,27 +19,58 @@
 
 package net.minecraftforge.common.brewing;
 
+import net.minecraft.world.Container;
 import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Recipe;
+import net.minecraft.world.item.crafting.RecipeType;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.common.ForgeMod;
+import org.apache.commons.lang3.NotImplementedException;
 
-public interface IBrewingRecipe {
+public interface IBrewingRecipe extends Recipe<Container>
+{
 
     /**
-     * Returns true is the passed ItemStack is an input for this recipe. "Input"
+     * Returns true is the passed ItemStack is an base for this recipe. "Input"
      * being the item that goes in one of the three bottom slots of the brewing
      * stand (e.g: water bottle)
      */
     boolean isInput(ItemStack input);
 
     /**
-     * Returns true if the passed ItemStack is an ingredient for this recipe.
+     * Returns true if the passed ItemStack is an reagent for this recipe.
      * "Ingredient" being the item that goes in the top slot of the brewing
      * stand (e.g: nether wart)
      */
     boolean isIngredient(ItemStack ingredient);
 
     /**
-     * Returns the output when the passed input is brewed with the passed
-     * ingredient. Empty if invalid input or ingredient.
+     * Returns the result when the passed base is brewed with the passed
+     * reagent. Empty if invalid base or reagent.
      */
     ItemStack getOutput(ItemStack input, ItemStack ingredient);
+
+    @Override
+    default boolean matches(Container container, Level level)
+    {
+        throw new NotImplementedException("Use the methods of IBrewingRecipe instead");
```

Comment:
Problematic code implementation, should not use `RecipeManager` if you're gonna go the half separated route. Since it is a data driven implementation though, you need to implement these properly.